SIS_import/sis_upload.rb

```diff
@@ -4,93 +4,151 @@
 require "fileutils"
 require "rubygems"
 require "nokogiri"
+require "digest"
 
 
-def upload_to_canvas(fileName, token, server, outputDirectory)
+def upload_to_canvas(currentDirectory, fileName, token, server, outputDirectory, outputFile, output_file_base_name)
 
-	# get file names
-	currentFileBaseName = File.basename(fileName)
-	currentFileBaseNameWithoutExtension = File.basename(fileName, ".zip")
 
-
-	#set the error flag, default to be false
+	# set the error flag, default to be false
 	upload_error = false
 
-	#open the output file
-	begin
-		outputFile = File.open(outputDirectory + currentFileBaseNameWithoutExtension + ".txt", "w")
+	# prior to upload current zip file, make an attempt to check the prior upload, whether it is finished successfully
+	if (prior_upload_error(currentDirectory, server, token))
+		## check first about the environment variable setting for MAILTO '
+		return "Previous upload job has not finished yet."
+	end
 
-		# Web Service call
-		p fileName
-		p server
+	# upload start time
+	outputFile.write("upload start time : " + Time.new.inspect)
 
-		json_data=`curl -H "Content-Type: application/zip" --data-binary @#{fileName} -H "Authorization: Bearer #{token}" #{server}/api/v1/accounts/1/sis_imports.json?import_type=instructure_csv`
+	# continue the current upload process
 
-		outputFile.write("#{json_data}\n")
-		parsed = parseJson(json_data)
+	# Web Service call
+	p fileName
+	p server
 
-		if (parsed["errors"])
-			## break and print error
-			error_array=parsed["errors"]
-			## hashmap ["message"=>"error_message"
-			uploadError = error_array[0]["message"]
-			outputFile.write("upload error: " + uploadError)
-			outputFile.write("\n")
+	json_data=`curl -H "Content-Type: application/zip" --data-binary @#{fileName} -H "Authorization: Bearer #{token}" #{server}/api/v1/accounts/1/sis_imports.json?import_type=instructure_csv`
```

Comment:
This is in my SIS upload project, not related to Pushyami's Section Management Tool. 

I have replaced the curl call in sis_update_url.rb file, but have not changed the sis_upload.rb yet. I will do that shortly, and open a separate Pull Request for it.